modules/data.remote/inst/SMAPvsModel_Comparison.R

```diff
@@ -0,0 +1,353 @@
+  
+  return(vtr)
+}
+
+
+#' @title runAnalysis
+#' 
+#' @description Runs multiple analyses on the specified site during a given date range
+#' and has the functionality to compare those values to actual soil moisture values
+#' collected from SMAP
+#'
+#' @param betySiteId bety site id for the desired location as (String)
+#' @param compare (boolean) to turn on/off comparison to actual smap data
+#'
+#' @return returns a list consisting of all of the analysis plots and statistics
+#'
+#' @examples 
+#' var <- runAnalysis('1000004945', '2016-01-02', '2016-07-16', TRUE)
+#' This runs an analysis on the Harvard Forest (1000004945) site with results between
+#' the date range of '2016-01-02' and '2016-07-16'
+#' 
+runAnalysis <- function(betySiteId, compare = TRUE){
+  
+  moistureFrame <- extractEnsambles(betySiteId)
+  
+  ## Grouping data by Date ##
+  grouped_ENS <- moistureFrame %>% 
+    group_by(time) %>% 
+    mutate(max = max(moisture)) %>%
+    mutate(min = min(moisture))
+  
+  ## Calculating Mean Per Ensemble ##
+  mean_per_ENS <- summarize(grouped_ENS, mean = mean(moisture))
+  
+  ## Calculating Standard Deviation ##
+  sd_per_ENS <- summarize(grouped_ENS, sd = sd(moisture))
+  
+  ## Calculating .025 and .975 Quantiles ##
+  quantiles_lower <- summarize(grouped_ENS, q_lower = quantile(moisture, .025))
+  quantiles_upper <- summarize(grouped_ENS, q_upper = quantile(moisture, .975))
+  
+  ## Hex Data Distribution with Mean-line Highlighted ##
+  p1 <- ggplot(moistureFrame, aes(x = time, y = moisture)) + geom_hex() + theme_bw() + 
+    guides(fill=guide_legend(title = 'Frequency of Model Data')) + 
+    theme(plot.title = element_text(hjust = 0.5)) +
+    ggtitle('Hex Distribution of Model Data') +
+    geom_line(data = mean_per_ENS, aes(x = time, y = mean, color = 'Mean of Model Data')) + 
+    labs(colour = '') + ylab('Surface Soil Moisture (ssm)') + xlab('Date of Prediction') +
+    labs(colour = 'Legend')
+  print(p1)
+  
+  ## Min/Max Values with 95% Quantile Range ## Still has messed up color legend
+  p2 <- ggplot(grouped_ENS, aes(x = time, y = moisture)) + theme_bw() + 
+    ggtitle('Daily Min and Max Values with 95% Quantile Range in Black') +
+    theme(legend.position = c(0.87, 0.75),
+          legend.background = element_rect(fill = "white", color = "black"),
+          plot.title = element_text(hjust = 0.5)) +
+    geom_point(data = grouped_ENS, aes(x = time, y = max, color = 'Daily Max')) +
+    geom_point(data = grouped_ENS, aes(x = time, y = min, color = 'Daily Min')) +
+    geom_line(data = mean_per_ENS, aes(x = time, y = mean, color = 'Mean')) +
+    geom_line(data = quantiles_lower, aes(x = time, y = q_lower)) +
+    geom_line(data = quantiles_upper, aes(x = time, y = q_upper)) + 
+    xlab('Date of Prediction') + ylab('Surface Soil Moisture (ssm)') +
+    labs(colour = 'Legend')
+  print(p2)
+  
+  if (compare) {
+    ## Analyses Requiring Actual SMAP Data ##
+    smap_data <- all_smap_data[[betySiteId]]
```

Comment:
all_smap_data appears to be a global variable. better to pass as a function argument, which could replace `compare` (i.e. make the default argument NULL and if it's not null then it's the data set you should compare to)